app/src/main/java/com/example/messenger/usecase/LoadMessagesUseCase.kt

```diff
@@ -0,0 +1,43 @@
+package com.example.messenger.usecase
+
+import com.example.messenger.event.ChattingRoomEvent
+import com.example.messenger.repository.message.MessageRepositoryImpl
+import io.reactivex.disposables.CompositeDisposable
+
+/**
+ * @author bsgreentea
+ */
+class LoadMessagesUseCase(
+    private val messageRepository: MessageRepositoryImpl,
+    private val disposables: CompositeDisposable
+) {
+
+    fun loadMessages(roomId: String, from: Int) {
+        loadLatestFiftyMessages(roomId, from)
+    }
+
+    private fun loadLatestFiftyMessages(roomId: String, from: Int) {
+        disposables.add(
+            messageRepository.getLatestFiftyMessages(roomId, from)
+                .doOnSuccess {
+                    for (message in it) {
+                        ChattingRoomEvent.addMessageToList(message)
+                    }
+                }
+                .subscribe()
+        )
+    }
```

Comment:
private fun loadLatestFiftyMessages(roomId: String, from: Int) {
        disposables.add(
            messageRepository.getLatestFiftyMessages(roomId, from)
                .doOnSuccess {
                    it.forEach { message ->
                        ChattingRoomEvent.addMessageToList(message)
                    }
                }
                .subscribe()
        )
    }

Reply:
```suggestion
    }
````` private fun loadLatestFiftyMessages(roomId: String, from: Int) {
        disposables.add(
            messageRepository.getLatestFiftyMessages(roomId, from)
                .doOnSuccess {
                    it.forEach { message ->
                        ChattingRoomEvent.addMessageToList(message)
                    }
                }
                .subscribe()
        )
    }`suggestion
    }
```

Reply:
이렇게 수정 가능

Reply:
foreach 사용